src/tokenize.js

```diff
@@ -202,35 +233,93 @@ function tokenize(source) {
                 if (++offset === length)
                     return null;
             }
+
             if (charAt(offset) === "/") {
-                if (++offset === length)
+                if (++offset === length) {
                     throw illegal("comment");
-                if (charAt(offset) === "/") { // Line
-                    isDoc = charAt(start = offset + 1) === "/";
-                    while (charAt(++offset) !== "\n")
-                        if (offset === length)
-                            return null;
-                    ++offset;
-                    if (isDoc) /// Comment
-                        setComment(start, offset - 1);
-                    ++line;
-                    repeat = true;
-                } else if ((curr = charAt(offset)) === "*") { /* Block */
-                    isDoc = charAt(start = offset + 1) === "*";
-                    do {
-                        if (curr === "\n")
-                            ++line;
-                        if (++offset === length)
-                            throw illegal("comment");
-                        prev = curr;
-                        curr = charAt(offset);
-                    } while (prev !== "*" || curr !== "/");
-                    ++offset;
-                    if (isDoc) /** Comment */
+                }
+                if (!alternateCommentMode) {
+                    // standard comment parsing
+                    if (charAt(offset) === "/") { // Line
+                        // check for triple-slash comment
+                        isDoc = charAt(start = offset + 1) === "/";
```

Comment:
What do you think of doing something like

```js
isDoc = (charAt(start = offset) === "/" && ++offset) || alternateCommentMode;
```

here so it always handles `///` and advances properly, and otherwise falls back to `//` in alternateCommentMode? Just asking because it seems that duplicating this section could potentially be avoided.

I'd say it would be fine if `///` in non-alternateCommentMode also coalesces consecutive lines.

Reply:
Thanks for looking. I agree that avoiding duplication would be good, however I am hesitant to change existing behavior in case other consumers of this library depend on it.

In the spirit of your comment, I consolidated the `/*` handling for original and alternate parsing modes. The `//`/`///` handling is a bit too divergent to consolidate further.